Move to URL instead of CURL

ethanmoss1 opened the pull request:
Bumped version to 2 as this requires a different library and no need for CURL.
Fixes issue #1, This stops package.el, elpaca and straight from trying to find an emacs package called curl.

If we want i can tackle #2 ?

ethanmoss1 committed:
FIX: Remove unnecessary dependency

Curl is not an emacs package, this is removed.

ethanmoss1 committed:
FIX: Added URL as dependency

ethanmoss1 committed:
FIX: need URL required

ethanmoss1 committed:
FIX: moved from Curl process -> Emacs URL Lib

ethanmoss1 committed:
VERSION BUMP

shombando commented:
@ethanmoss1 thanks for putting this together and making it an Emacs only package, I love that. I won't be able to test it for a couple days but I'm on-board with merging it and going this direction. I'll test and merge ASAP. Thanks again!

ethanmoss1 commented:
Oh no, thank you! Especially for doing most of the work! It was fun learning the URL package/lib. I might tackle some of the long-term goals soon, so I'll create a pull request when I've had a go.

Any issues just shout and I'll get them changed ready for merging.

ethanmoss1 committed:
Added Autoloads

ethanmoss1 committed:
Removed package Requirement

This upsets Elpaca as it trys to install url

ethanmoss1 committed:
ntfy.el: Added main group

ethanmoss1 committed:
ntfy.el: added input checker

ethanmoss1 committed:
ntfy.el: some text changes

ethanmoss1 committed:
ntfy.el: add the ability to customise priority

ethanmoss1 committed:
ntfy.el: ntfy--publish-message

The function can now check inputs as well as change priority on the message.

ethanmoss1 committed:
ntfy.el: bugfix priority

The priority should be 1 to 5. and now the checks use a known good string or number

ethanmoss1 committed:
FIX: naming changes and more readme examples

ethanmoss1 committed:
FEAT: Android intent and Scheduled delivery

ethanmoss1 committed:
First draft of the json implementation

ethanmoss1 committed:
FEAT: emoji-selector in interactive function

ethanmoss1 committed:
FEAT: Added interactive way to change tags with emojis!

ethanmoss1 committed:
Updated docstrings and commentary

ethanmoss1 committed:
Split out functions, added attach functionality

ethanmoss1 committed:
added possible new changes to the example file

ethanmoss1 committed:
Start of URL branch

ethanmoss1 committed:
FIX: ntfy.el - Re-added the publish message function

Also fixed some comments and bugs
Added a check on the inputs

ethanmoss1 committed:
FIX: ntfy.el - Messages and bug fixes

ntfy--publish-message: work now with the tags
ntfy-message-*: these functions all have the keys removed
ntfy--check-inputs: regex updated

ethanmoss1 committed:
readme: updated improvements and goals

ethanmoss1 committed:
ntfy.el: updated defaults

ethanmoss1 committed:
test.el: added very basic test.

ethanmoss1 committed:
ntfy-compilation.el: added new minor mode for compilation

ethanmoss1 committed:
Merge branch 'no-json'

shombando commented on ntfy-example.org:
```diff
@@ -0,0 +1,161 @@
+  (setopt ntfy-server "http://ntfy.domain.com"
+		  ntfy-topic "emacs"
+		  ntfy-title "Notification from emacs"
+		  ntfy-tags "purple_circle,loudspeaker"
```
@ethanmoss1 I think this needs to be 
```
		ntfy-tags '("purple_circle" "loudspeaker")
```

shombando commented on readme.org:
```diff
@@ -9,20 +9,29 @@ The =ntfy.el= file can be downloaded and loaded via =load-file= and other instal
+    :config (setopt ntfy-server "https://ntfy.sh"
+  		          ntfy-topic "mytopic"
+  		          ntfy-header "Notification from emacs"
+  		          ntfy-tags "purple_circle,loudspeaker"
```
Same change here:
```
		ntfy-tags '("purple_circle" "loudspeaker")
```

shombando commented on ntfy.el:
```diff
@@ -1,82 +1,186 @@
+
+
+;;;--- Internal Functions
+(defun ntfy--interactive-emoji-selector ()
```
@ethanmoss1 this is really nice functionality, love it. I do wonder though if it makes sense to maintain an emoji list in the project that will be out of sync with both ntfy and emacs? Is there a way to invoke emacs built-in `emoji-search`?

ethanmoss1 replied:
Looking at emacs' provided `emoji.el` and the script `scripts/emoji-convert.sh` from ntfy, it could take the emoji selection code from emacs and use the emoji.json file in the ntfy to get the alias use for ntfy dynamically.

This will keep it inline if for any reason ntfy extends its emoji list.

shombando reviewed (changes requested):
@ethanmoss1 this is so great! You have totally surpassed my original foundation. I have a couple requests for documentation updates (needed before merge) and a suggestion around emoji-list (for consideration, not merge blocker).
I'd like to invite you to be a maintainer of this project if you're willing. After this merge is done, I'll update the docs to point to Github as the canonical source of the package so you can make merges as a maintainer here.
Thanks for all the great updates and your immense patience.

ethanmoss1 committed:
ntfy-example.org: correct config, updated whitespace

ethanmoss1 committed:
readme.org: correct config, updated whitespace

ethanmoss1 committed:
readme.org: OS list expanded

ethanmoss1 committed:
Merge branch 'no-json'

Still was using the 'no-json' branch, back to main.
